workers/unity/Packages/com.improbable.gdk.buildsystem/Configuration/SpatialOSBuildConfigurationEditor.cs

```diff
@@ -73,81 +92,408 @@ public override void OnInspectorGUI()
-                    .Select(sceneAsset => new SceneItem(sceneAsset, false, scenesInAssetDatabase))
-                    .ToList();
 
-                var horizontalLayout = Screen.width > ScreenWidthForHorizontalLayout;
+            var newRect = new Rect(minX, minY, maxX - minX, maxY - minY);
 
-                using (horizontalLayout ? new EditorGUILayout.HorizontalScope() : null)
+            return newRect;
+        }
+
+        private static Rect RectExpand(Rect a, Vector2 amount)
+        {
+            return new Rect(a.xMin - amount.x, a.yMin - amount.y, a.width + amount.x * 2, a.height + amount.y * 2);
+        }
+
+        private class ScopedGUIColor : IDisposable
+        {
+            public ScopedGUIColor(Color color)
+            {
+                oldColor = GUI.color;
+                GUI.color = color;
+            }
+
+            public void Dispose()
+            {
+                GUI.color = oldColor;
+            }
+
+            private readonly Color oldColor;
+        }
+
+        private void DrawScenesInspectorForWorker(WorkerBuildConfiguration configurationForWorker)
+        {
+            EditorGUILayout.LabelField("Scenes", EditorStyles.boldLabel);
+            var workerControlId = GUIUtility.GetControlID(FocusType.Passive);
+            var dragState = (DragAndDropInfo) GUIUtility.GetStateObject(typeof(DragAndDropInfo), workerControlId);
+            var currentEventType = Event.current.type;
+
+            using (IndentLevelScope(1))
+            {
+                using (var check = new EditorGUI.ChangeCheckScope())
                 {
-                    using (horizontalLayout ? new EditorGUILayout.VerticalScope() : null)
+                    using (new EditorGUILayout.HorizontalScope())
                     {
                         EditorGUILayout.LabelField("Scenes to include (in order)");
-                        DrawSceneList(scenesToShowInList, true, true);
+                        GUILayout.FlexibleSpace();
+                        if (GUILayout.Button(AddSceneButtonContents, EditorStyles.miniButton))
+                        {
+                            CurrentObjectPickerWindowId = GUIUtility.GetControlID(FocusType.Passive);
+                            EditorGUIUtility.ShowObjectPicker<SceneAsset>(null, false, "t:Scene",
+                                CurrentObjectPickerWindowId);
+                        }
+
+                        HandleObjectSelectorUpdated(configurationForWorker);
                     }
 
-                    using (horizontalLayout ? new EditorGUILayout.VerticalScope() : null)
+                    if (configurationForWorker.ScenesForWorker.Length == 0)
                     {
-                        using (horizontalLayout ? IndentLevelScope(-EditorGUI.indentLevel) : null)
+                        // Allow dropping to form a new list.
+                        EditorGUILayout.HelpBox("No scenes", MessageType.Info);
```

Comment:
Message is a bit unclear. "No scene in project", "No scenes found" ?

Reply:
Changed to "Add scenes here"
Should this also fail validation? If there's no scene, a worker can't really do anything.